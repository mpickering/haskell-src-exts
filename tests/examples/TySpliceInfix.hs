{-# LANGUAGE TemplateHaskell #-}
a :: $(f $ g $ x)
a = ff $ gg $ xx
