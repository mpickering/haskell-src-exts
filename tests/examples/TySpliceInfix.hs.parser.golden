ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 1 1 4 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/TySpliceInfix.hs" 1 1 1 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 2 1 2 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 2 1 2 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 2 1 2 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 3 1 3 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 4 1 4 1
            , SrcSpan "tests/examples/TySpliceInfix.hs" 4 1 4 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 1 1 1 33
            , srcInfoPoints =
                [ SrcSpan "tests/examples/TySpliceInfix.hs" 1 1 1 13
                , SrcSpan "tests/examples/TySpliceInfix.hs" 1 30 1 33
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 1 14 1 29
                , srcInfoPoints = []
                }
              "TemplateHaskell"
          ]
      ]
      []
      [ TypeSig
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 1 2 18
            , srcInfoPoints =
                [ SrcSpan "tests/examples/TySpliceInfix.hs" 2 3 2 5 ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 1 2 2
                , srcInfoPoints = []
                }
              "a"
          ]
          (TySplice
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 6 2 18
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/TySpliceInfix.hs" 2 6 2 8
                   , SrcSpan "tests/examples/TySpliceInfix.hs" 2 17 2 18
                   ]
               }
             (ParenSplice
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 6 2 18
                  , srcInfoPoints =
                      [ SrcSpan "tests/examples/TySpliceInfix.hs" 2 6 2 8
                      , SrcSpan "tests/examples/TySpliceInfix.hs" 2 17 2 18
                      ]
                  }
                (InfixApp
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 8 2 17
                     , srcInfoPoints = []
                     }
                   (Var
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 8 2 9
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 8 2 9
                           , srcInfoPoints = []
                           }
                         (Ident
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 8 2 9
                              , srcInfoPoints = []
                              }
                            "f")))
                   (QVarOp
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 10 2 11
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 10 2 11
                           , srcInfoPoints = []
                           }
                         (Symbol
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 10 2 11
                              , srcInfoPoints = []
                              }
                            "$")))
                   (InfixApp
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 12 2 17
                        , srcInfoPoints = []
                        }
                      (Var
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 12 2 13
                           , srcInfoPoints = []
                           }
                         (UnQual
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 12 2 13
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 12 2 13
                                 , srcInfoPoints = []
                                 }
                               "g")))
                      (QVarOp
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 14 2 15
                           , srcInfoPoints = []
                           }
                         (UnQual
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 14 2 15
                              , srcInfoPoints = []
                              }
                            (Symbol
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 14 2 15
                                 , srcInfoPoints = []
                                 }
                               "$")))
                      (Var
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 16 2 17
                           , srcInfoPoints = []
                           }
                         (UnQual
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 16 2 17
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 2 16 2 17
                                 , srcInfoPoints = []
                                 }
                               "x")))))))
      , PatBind
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 1 3 17
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 1 3 2
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 1 3 2
                  , srcInfoPoints = []
                  }
                "a"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 3 3 17
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/TySpliceInfix.hs" 3 3 3 4 ]
               }
             (InfixApp
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 5 3 17
                  , srcInfoPoints = []
                  }
                (Var
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 5 3 7
                     , srcInfoPoints = []
                     }
                   (UnQual
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 5 3 7
                        , srcInfoPoints = []
                        }
                      (Ident
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 5 3 7
                           , srcInfoPoints = []
                           }
                         "ff")))
                (QVarOp
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 8 3 9
                     , srcInfoPoints = []
                     }
                   (UnQual
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 8 3 9
                        , srcInfoPoints = []
                        }
                      (Symbol
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 8 3 9
                           , srcInfoPoints = []
                           }
                         "$")))
                (InfixApp
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 10 3 17
                     , srcInfoPoints = []
                     }
                   (Var
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 10 3 12
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 10 3 12
                           , srcInfoPoints = []
                           }
                         (Ident
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 10 3 12
                              , srcInfoPoints = []
                              }
                            "gg")))
                   (QVarOp
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 13 3 14
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 13 3 14
                           , srcInfoPoints = []
                           }
                         (Symbol
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 13 3 14
                              , srcInfoPoints = []
                              }
                            "$")))
                   (Var
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 15 3 17
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 15 3 17
                           , srcInfoPoints = []
                           }
                         (Ident
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/TySpliceInfix.hs" 3 15 3 17
                              , srcInfoPoints = []
                              }
                            "xx"))))))
          Nothing
      ]
  , []
  )
